pandas/tests/series/test_analytics.py

```diff
@@ -1489,7 +1490,8 @@ def test_value_counts_with_nan(self):
             "unicode_",
             "timedelta64[h]",
             pytest.param(
-                "datetime64[D]", marks=pytest.mark.xfail(reason="GH#7996", strict=True)
+                "datetime64[D]",
+                marks=pytest.mark.xfail(reason="GH#7996", strict=not PY35),
```

Comment:
I don't understand this xfail. Typically we just reference open issues. What's causing this to fail on 3.5? What's the failure you get?

Reply:
>What's causing this to fail on 3.5?

Just to be clear: The problem is that this test *doesn't* fail in Python 3.5.  But since it's marked with `xfail(..., strict=True)`, that breaks the test suite.

You can see a failed build log here:
https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=15986

The relevant lines are:
```
=================================== FAILURES ===================================
 TestCategoricalSeriesAnalytics.test_drop_duplicates_categorical_non_bool[True-datetime64[D]] 
[gw1] linux -- Python 3.5.3 /home/vsts/miniconda3/envs/pandas-dev/bin/python
[XPASS(strict)] GH#7996
 TestCategoricalSeriesAnalytics.test_drop_duplicates_categorical_non_bool[False-datetime64[D]] 
[gw1] linux -- Python 3.5.3 /home/vsts/miniconda3/envs/pandas-dev/bin/python
[XPASS(strict)] GH#7996
 TestCategoricalSeriesAnalytics.test_drop_duplicates_categorical_non_bool[None-datetime64[D]] 
[gw1] linux -- Python 3.5.3 /home/vsts/miniconda3/envs/pandas-dev/bin/python
[XPASS(strict)] GH#7996
-------- generated xml file: /home/vsts/work/1/s/test-data-multiple.xml --------
```

>What's the failure you get?

In Python 3.7, the test `xfails` as expected.  Removing `xfail` identifies [this line as the problem][1].  And here's what `pytest` shows in that case:

```
>   raise_assert_detail(obj, msg, lobj, robj)
E   AssertionError: Series are different
E
E   Series values are different (50.0 %)
E   [left]:  [False, True, True, True]
E   [right]: [False, False, False, True]

pandas/_libs/testing.pyx:178: AssertionError
```

[1]: https://github.com/pandas-dev/pandas/blob/11b28ea584b3647bc8ca5b56889c9199c0af5ec5/pandas/tests/series/test_analytics.py#L1504

Reply:
>I don't understand this xfail. Typically we just reference open issues.

The referenced issue seems to imply that the trouble is related to converting/comparing  `datetime[D]` to `datetime[ns]`.  In this test, the input is `datetime[D]`, but it's implicitly converted to `datetime[ns]` when it is loaded into a `Categorical`.

One simple hack to make this test pass is to change `datetime[D]` to `datetime[ns]`.  That doesn't seem appropriate, though.

Anyway, to get an idea of what is actually going wrong, here's what happens when I try the test's first three lines in my terminal.  Note that the first item becomes `NaT` for some reason.

```
In [190]: cat_array = np.array([1, 2, 3, 4, 5], dtype=np.dtype(dtype))

In [191]: input1 = np.array([1, 2, 3, 3], dtype=np.dtype(dtype))

In [192]: tc1 = Series(Categorical(input1, categories=cat_array, ordered=False))

In [193]: tc1
Out[193]:
0          NaT
1   1970-01-03
2   1970-01-04
3   1970-01-04
dtype: category
Categories (5, datetime64[ns]): [1970-01-02, 1970-01-03, 1970-01-04, 1970-01-05, 1970-01-06]
```